### PR TITLE
Feature public sentryevent.tags

### PR DESCRIPTION
**Pull request by samukce**
Complementing the PR #149. Some tests case to ensure public assign from SentryEvent.Tags

---
This change is [Reviewable](https://reviewable.io/reviews/getsentry/raven-csharp/159)

### CODE REVIEW

**Comment by samukce** [comment]
@asbjornu done. o/

**Comment by asbjornu** [comment]
@samukce Excellent! But can you perhaps remove the three merge commits from this PR? This is rather easy to do by resetting this branch to the latest `develop` and then doing `cherry-pick` to pick the interesting commit from here. Like so:

```shell
git fetch getsentry
git checkout feature/publicTagsSentryEvent
git reset --hard getsentry/develop
git cherry-pick 8d868bf5417ee8f5824faf1102d05306f4708278
git push --force
```

**Comment by samukce** [comment]
@asbjornu thanks for the feedback. Done.

File: src/tests/SharpRaven.UnitTests/Data/SentryEventTests.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            Assert.That(sentryEvent.Tags, Is.Not.Null);
+        }
+
+        [Test]
+        public void ShouldTheSameDictionaryFromConstructor()
+        {
+            var sentryEvent = new SentryEvent(new Exception("foo"));
+            var beforeDictionary = sentryEvent.Tags;
+
+            Assert.That(beforeDictionary, Is.SameAs(sentryEvent.Tags));
```
Comment (asbjornu) [inline src/tests/SharpRaven.UnitTests/Data/SentryEventTests.cs]: I don't quite see the value of this assertion.

Comment (samukce) [inline src/tests/SharpRaven.UnitTests/Data/SentryEventTests.cs]: The get/set execute some procedures how instance new dictionary. Some refactor could inject a bug, for instance, a new dictionary, what could losing data.

Comment (asbjornu) [inline src/tests/SharpRaven.UnitTests/Data/SentryEventTests.cs]: Good point!

File: src/tests/SharpRaven.UnitTests/Data/SentryEventTests.cs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        {
+            Assert.That(new SentryEvent(new Exception("foo")).Tags, Is.Not.Null);
+        }
+
+        [Test]
+        public void ShouldNotNullDictionaryWhenAssignNullForTags()
+        {
+            var sentryEvent = new SentryEvent(new Exception("foo")) { Tags = null};
+
+            Assert.That(sentryEvent.Tags, Is.Not.Null);
```
Comment (genne) [inline src/tests/SharpRaven.UnitTests/Data/SentryEventTests.cs]: Regarding this. The test case is great, but the "feature" feels a bit weird. I would really prefer if the setter of `Tags` threw an `ArgumentNullException` or similar when trying to set it to `null`. Setting something to `null` and then making sure it wasn't set to `null` doesn't really make sense?

Comment (asbjornu) [inline src/tests/SharpRaven.UnitTests/Data/SentryEventTests.cs]: @genne: Good point. Let's do `ArgumentNullException`, then. PR incoming? :smile:

Comment (samukce) [inline src/tests/SharpRaven.UnitTests/Data/SentryEventTests.cs]: I think that this api should work similar to the log4net, i mean when any is wrong with the API, this not interfere of the function of the software.

Comment (asbjornu) [inline src/tests/SharpRaven.UnitTests/Data/SentryEventTests.cs]: @samukce While I understand that design principle from a logging framework, there's already quite a few `throw` statements in there.

Comment (samukce) [inline src/tests/SharpRaven.UnitTests/Data/SentryEventTests.cs]: @asbjornu  but do you agree that some simple exceptions could be avoided?

Comment (asbjornu) [inline src/tests/SharpRaven.UnitTests/Data/SentryEventTests.cs]: Yes, but in this case, I'm basically ¯\_(ツ)_/¯.

**Pull request merged**